### PR TITLE
feat(linter/eslint): Implement default_case rule

### PR DESCRIPTION
**Pull request by jelly**
Rule Detail:
[link](https://eslint.org/docs/latest/rules/default-case)

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/default_case.rs [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+#[derive(Debug, Clone)]
+pub struct DefaultCaseConfig {
+    comment_pattern: Regex,
+}
+
+impl Default for DefaultCaseConfig {
+    fn default() -> Self {
+        Self {
+            comment_pattern: RegexBuilder::new(r"^no default$")
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/default_case.rs:27]: If the default is a simple string match, can we make this a fast path by defining

`comment_pattern: Option<Regex>`,

and just check for `no default` when it's `None`?

Constructing the regex and running it both introduce a small performance overhead.

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/eslint/default_case.rs (hunk 1 above)
Comment (jelly) [reply crates/oxc_linter/src/rules/eslint/default_case.rs]: Ah, that makes sense! I'm still learning Rust :)

I'm not sure if I pushed the most Rust idiomatic approach but the tests are happy.

**Comment by Boshen** [comment]
I pushed some changes:

* demonstrate the `None` case
* removed the allocation from `let default_cases = cases.iter().filter(|v| v.test.is_none()).collect::<Vec<_>>();`
* made the code more compact to demonstrate some nice Rust APIs

**Review by Boshen — APPROVED** [review]

**Comment by jelly** [comment]
> I pushed some changes:
> 
>     * demonstrate the `None` case
> 
>     * removed the allocation from `let default_cases = cases.iter().filter(|v| v.test.is_none()).collect::<Vec<_>>();`, the code no longer allocates any heap memory 😁
> 
>     * made the code more compact to demonstrate some nice Rust APIs

Thanks for the explanation!

**Pull request merged**